src/utils/dlck/cmds/dlck_cmd_pool_check.c

```diff
@@ -0,0 +1,123 @@
+/**
+ * (C) Copyright 2025 Hewlett Packard Enterprise Development LP
+ *
+ * SPDX-License-Identifier: BSD-2-Clause-Patent
+ */
+
+#include <daos/mem.h>
+#include <daos_srv/vos.h>
+#include <daos_srv/dlck.h>
+#include <daos_srv/mgmt_tgt_common.h>
+
+#include "../dlck_args.h"
+#include "../dlck_engine.h"
+#include "../dlck_pool.h"
+
+static int
+exec_one(struct xstream_arg *xa, int idx)
+{
+	struct dlck_file *file;
+	char             *path;
+	daos_handle_t     poh;
+	int               rc;
+
+	xa->xs->tgt_id = idx; /** pretend to be a target of the requested ID */
+
+	rc = dlck_engine_xstream_init(xa->xs);
+	if (rc != DER_SUCCESS) {
+		return rc;
+	}
+
+	d_list_for_each_entry(file, &xa->ctrl->files.list, link) {
+		/** do not process the given file if the target is excluded */
+		if ((file->targets_bitmap & (1 << idx)) == 0) {
+			continue;
+		}
+
+		rc = ds_mgmt_file(xa->ctrl->engine.storage_path, file->po_uuid, VOS_FILE, &idx,
+				  &path);
+		if (rc != DER_SUCCESS) {
+			break;
+		}
+
+		DLCK_PRINT(&xa->ctrl->print, "\n");
+		rc = vos_pool_open_metrics(path, file->po_uuid, DLCK_POOL_OPEN_FLAGS, NULL,
+					   &xa->ctrl->print, &poh);
+		if (rc == DER_SUCCESS) {
```

Comment:
What is the main action after opening the pool? Just check whether the pool can be opened or not? For `pool_check` case, that is fine. But as my understand, exec_one() is the general interface for most of DLCK sub command.